### ISSUE TITLE
Bug: Manage subscription link in Action bar is incorrect

### ISSUE THREAD

**Issue by crisbusquets** [Automattic/wp-calypso#86454]
### Current flow

https://github.com/Automattic/wp-calypso/assets/25607244/115ccf77-7f39-46b7-9fb2-fd02d3f6bc72

### Proposal
- Change `Manage subscriptions` to `Manage subscription` in the Action bar
- Open `https://wordpress.com/read/subscriptions/{id}` instead of `https://wordpress.com/read/subscriptions?s={site}`. It saves a click and it opens what the user needs

**Comment by cat-og** [comment]
📌 REPRODUCTION RESULTS
- Tested & Replicated

📌 FINDINGS/SCREENSHOTS/VIDEO
- This appears to be deliberate, as the individual subscription can still be updated from that control bar, with the same options that you would see on the individual subscription page. 

![Image](https://github.com/Automattic/wp-calypso/assets/86028337/563e31d8-6f2c-456f-b784-9461b67693a9)

@crisbusquets Did you want this link to the individual subscription page to be in addition to the subscription management options in the toolbar?

I do think that it is worth it to have a direct link to where users can view all the sites they are subscribed to, but I do agree that the wording could be more clear, whichever way this goes. 

If we decide to keep the current option to link to all Subscription options, that could be updated to 'Manage All Subscriptions' to keep it seperate from the current subscription option for the site the user is now on. 


📌 ACTIONS
- Requested author feedback

**Comment by crisbusquets** [comment]
> @crisbusquets Did you want this link to the individual subscription page to be in addition to the subscription management options in the toolbar?

Hi @cat-og !

I was referring to the `Manage subscriptions` link that appears when a user clicks on the three-dots icon in the Action Bar (not the "Subscribed" link).

<img width="441" alt="Screenshot 2024-01-22 at 09 35 57" src="https://github.com/Automattic/wp-calypso/assets/25607244/4795fb89-53f5-4e95-aacb-30305e290404">

In this example, the blog is called Phusions. Since the options are under the blog's name, I'd assume that all the links are relative to this.

This means that instead of opening this:
<img width="1310" alt="Screenshot 2024-01-22 at 09 37 23" src="https://github.com/Automattic/wp-calypso/assets/25607244/caeb626b-ab20-41ad-b7cb-6864a10f850d">

We should open this:
<img width="1167" alt="Screenshot 2024-01-22 at 09 38 16" src="https://github.com/Automattic/wp-calypso/assets/25607244/58ed7ad4-12dd-40bc-a6c8-0a8df3dbd0d9">

And also change the button's text to `Manage subscription`, in singular.

What do you think?

**Comment by pkuliga** [comment]
@crisbusquets what if the user is not logged in? In this case, the "Manage subscriptions" link takes me to the https://wordpress.com/email-subscriptions/ site.

**Comment by crisbusquets** [comment]
> @crisbusquets what if the user is not logged in? In this case, the "Manage subscriptions" link takes me to the https://wordpress.com/email-subscriptions/ site.

If they're not logged in the interface changes a bit (see screenshot below). 

<img width="564" alt="Screenshot 2024-03-11 at 09 28 32" src="https://github.com/Automattic/wp-calypso/assets/25607244/61d8414b-fc1b-4c8c-b5ee-ed9c9cb06473">

In this case, it makes sense that we're redirecting them to `https://wordpress.com/email-subscriptions/`, so they can log in and manage their subscription.

After logging in, we also redirect them to `https://wordpress.com/read/subscriptions` instead of the individual subscription page.

If they've initiated the flow in a specific blog, I'd assume they want to manage their subscription to that specific blog.

### If they aren't subscribed
In the case where I'm not logged in, and I'm also not subscribed to that site, then we can redirect them to the generic management page (`https://wordpress.com/read/subscriptions`).

Does it make sense, or am I adding too much complexity?

**Comment by inaikem** [comment]
Confirmed that the link still lands on `https://wordpress.com/read/subscriptions?s=$SITE_SLUG` vs `https://wordpress.com/read/subscriptions/$ID`.

@simison, nudging to ask if you have a sense of when this might be worked on and a suggestion `[Pri]` label.

**Comment by simison** [comment]
@inaikem the plan is to remove the action bar next: p58i-ioo-p2#comment-64814

**Comment by inaikem** [comment]
Thank you! Do we have an idea of a timeline on this and how many times this "Manage Subscription" link is being clicked across all users?

I'd generally characterise any link that doesn't land on the expected page as a blocker issue

**Comment by annezazu** [comment]
This plan to remove has not moved forward so this issue remains.

**Comment by inaikem** [comment]
Thanks Anne! Noting here that I saw your latest P2 comment on this (pfYzsZ-On-p2#comment-1117) and the linked issue above  (#98042).

**Comment by aaronrobertshaw** [comment]
I've had a look into a fix for this. 

Renaming the links and updating the URLs for the contexts outlined in this issue should be easy enough. That said, I'm having a hard time getting context and background around why the [current links](fbhepr%2Skers%2Sjcpbz%2Sjc%2Qpbagrag%2Szh%2Qcyhtvaf%2Snpgvbaone.cuc%3Se%3Q3431422n%23635%2Q642-og) are to `following/manage` and whether they need to be kept that way.

From what I can tell the Manage Subscription/s links currently point to `following/manage` for stat tracking, and backwards compatibility purposes with the [subscription mailer](fbhepr%2Skers%2Sjcpbz%2Sjc%2Qpbagrag%2Szh%2Qcyhtvaf%2Srznvy%2Qfhofpevcgvbaf%2Sfhofpevcgvba%2Qznvyre.cuc%3Se%3Q62p733q7-og). It'd be great if someone could help clarify or confirm that though. 

@tyxla or @jsnajdr I saw you [updated](https://github.com/Automattic/wp-calypso/pull/86410) the `following/manage` controller a while back, do you have any insights you can share here? 🙏 

If we only need to continue supporting the `following/manage` route for links in older subscription emails, I was wondering if we could simply update these manage subscription links in the floating actionbar to use the correct `read/subscriptions` and `read/subscription/:sub_id` paths?

This area is all completely foreign to me, so any nudges in the right direction would be hugely appreciated, thanks! 🙇

**Comment by tyxla** [comment]
Thanks for working on it @aaronrobertshaw!

> Renaming the links and updating the URLs for the contexts outlined in this issue should be easy enough. That said, I'm having a hard time getting context and background around why the current links are to following/manage and whether they need to be kept that way.

The `/following/manage` paths were removed a while ago, in favor of the `/read/subscription` ones. So it makes sense to use the latter ones. The `/following/manage` paths remain only for compatibility nowadays and they will redirect to the `/read/subscriptions`, see:

https://github.com/Automattic/wp-calypso/blob/c58ca26af6fc8bd613bcb5bf6ba84bfad0fd8b66/client/reader/following/index.js#L7

and:

https://github.com/Automattic/wp-calypso/blob/c58ca26af6fc8bd613bcb5bf6ba84bfad0fd8b66/client/reader/following/controller.js#L6

> If we only need to continue supporting the following/manage route for links in older subscription emails, I was wondering if we could simply update these manage subscription links in the floating actionbar to use the correct read/subscriptions and read/subscription/:sub_id paths?

That would make sense to me 👍 I don't see a good reason not to do that.

**Comment by aaronrobertshaw** [comment]
Appreciate the clarification and confirmation @tyxla 🙇 

I also wasn't sure about the stats side of things, but on a closer look, I can see the stat bump is based off the click rather than a request to a specific URL. 

With all that out of the way, I'll get a PR up shortly.

**Comment by aaronrobertshaw** [comment]
There's a draft PR available in 170181-ghe-Automattic/wpcom. It's not there yet but it's a start.

**Comment by aaronrobertshaw** [comment]
After playing around with trying to redirect a previously logged out but subscribed user to the individual subscription page, I've decided to punt that from 170181-ghe-Automattic/wpcom. 

There are a couple of flows here that would all need the blog id passed through to them to behave consistently. The email content might also need a tweak to align the title with the body text. More details over in: 170181-ghe-Automattic/wpcom#discussion_r163815

The TL;DR is, this suggested redirect might be better revisited during an effort to improve the /email-subscriptions flow. This would also help keep momentum on the current [improvements to the floating action bar](https://github.com/Automattic/wp-calypso/issues/98042).

**Comment by inaikem** [comment]
Thanks @aaronrobertshaw. @simison, would improvements to the /email-subscriptions flow connect with Zap's current remit? I'm trying to figure out the bext team to point this to for now.

**Comment by jeherve** [comment]
I think this is good to close for now since the immediate issue is closed. We can always revisit the /email-subscriptions flow improvements when we next iterate on that part of Subscriptions.
